llama_index/storage/chat_store/simple_chat_store.py

```diff
@@ -0,0 +1,72 @@
+import json
+import os
+from typing import Dict, List, Optional
+
+import fsspec
+
+from llama_index.bridge.pydantic import Field
+from llama_index.llms import ChatMessage
+from llama_index.storage.chat_store.base import BaseChatStore
+
+
+class SimpleChatStore(BaseChatStore):
+    """Simple chat store."""
+
+    store: Dict[str, List[ChatMessage]] = Field(default_factory=dict)
+
+    @classmethod
+    def class_name(cls) -> str:
+        """Get class name."""
+        return "SimpleChatStore"
+
+    def set_messages(self, key: str, messages: List[ChatMessage]) -> None:
+        """Set messages for a key."""
+        self.store[key] = messages
+
+    def get_messages(self, key: str) -> List[ChatMessage]:
+        """Get messages for a key."""
+        return self.store.get(key, [])
+
+    def add_message(self, key: str, message: ChatMessage) -> None:
+        """Add a message for a key."""
+        self.store.setdefault(key, []).append(message)
+
+    def delete_messages(self, key: str) -> None:
+        """Delete messages for a key."""
+        self.store.pop(key, None)
+
+    def undo_last_message(self, key: str) -> Optional[ChatMessage]:
```

Comment:
Would there be any reason to include the means to delete any given message from a list of messages? I'm thinking for example if chat history is display in some front-end and maybe you want to delete a message that's somewhere in the middle or even beginning of chat history?